### PR TITLE
[anaconda] Python (Pip) Security Update for pyarrow (GHSA-5wvp-7f3h-6wmm)

### PR DESCRIPTION
**Pull request by bhupendra-vaishnav**
**Dev container name:** 

anaconda

**Description:**

This PR patches the following vulnerability:

[GHSA-5wvp-7f3h-6wmm](https://github.com/advisories/GHSA-5wvp-7f3h-6wmm) - related to the pyarrow package;
This vulnerability comes from the continuumio/anaconda3 image used upstream for the anaconda devcontainer.

**Changelog:**

Updated Dockerfile & locked version for following package :
- pyarrow - minimum package version set to 14.0.1.

Other changes to support primary change :
- Updated tests to verify pyarrow minimum version 14.0.1 to fix [GHSA-5wvp-7f3h-6wmm](https://github.com/advisories/GHSA-5wvp-7f3h-6wmm).

**Checklist:**

 Checked that applied changes work as expected.

### CODE REVIEW

**Comment by bhupendra-vaishnav** [comment]
@microsoft-github-policy-service agree company="GitHub"

**Comment by bhupendra-vaishnav** [comment]
@microsoft-github-policy-service agree company="GitHub"

**Comment by bhupendra-vaishnav** [comment]
@microsoft-github-policy-service agree company="GitHub"

**Comment by bhupendra-vaishnav** [comment]
@microsoft-github-policy-service agree company="GitHub"

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,612 @@
+{
```
Comment (eljog) [inline package-lock.json:1]: Did any command that you had to run for the building/testing for this feature required to build the tool, that caused the generation of this lock file?
I think the corresponding package.json belongs to the build tools. Not sure why these lock file was not checked in ever.
Can you help me understand the process you followed?

Comment (bhupendra-vaishnav) [reply package-lock.json]: Removed package-lock.json file as its not require

**Review by eljog — APPROVED** [review]

**Pull request merged**